packages/salesforcedx-vscode-core/test/vscode-integration/commands/functions/forceFunctionStop.test.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { CliCommandExecutor } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as path from 'path';
+import { assert, createSandbox, SinonSandbox, SinonStub } from 'sinon';
+import { Uri } from 'vscode';
+import { channelService } from '../../../../src/channels';
+import {
+  forceFunctionStart,
+  ForceFunctionStartExecutor
+} from '../../../../src/commands/functions/forceFunctionStart';
+import { forceFunctionStop } from '../../../../src/commands/functions/forceFunctionStop';
+import { nls } from '../../../../src/messages';
+import { notificationService } from '../../../../src/notifications';
+import { telemetryService } from '../../../../src/telemetry';
+import { getRootWorkspacePath } from '../../../../src/util';
+import { MockExecution } from './mockExecution';
+
+describe('Force Function Stop', () => {
+  let sandbox: SinonSandbox;
+  let cliCommandExecutorStub: SinonStub;
+  const channelServiceStubs: {
+    [key: string]: SinonStub;
+  } = {};
+  const notificationServiceStubs: {
+    [key: string]: SinonStub;
+  } = {};
+  const telemetryServiceStubs: {
+    [key: string]: SinonStub;
+  } = {};
+  let hrtimeStub: SinonStub;
+  beforeEach(() => {
+    sandbox = createSandbox();
+
+    cliCommandExecutorStub = sandbox.stub(
+      CliCommandExecutor.prototype,
+      'execute'
+    );
+    channelServiceStubs.appendLineStub = sandbox.stub(
+      channelService,
+      'appendLine'
+    );
+    notificationServiceStubs.showSuccessfulExecutionStub = sandbox.stub(
+      notificationService,
+      'showSuccessfulExecution'
+    );
+    notificationServiceStubs.showSuccessfulExecutionStub.returns(
+      Promise.resolve()
+    );
+    notificationServiceStubs.showWarningMessageStub = sandbox.stub(
+      notificationService,
+      'showWarningMessage'
+    );
+    telemetryServiceStubs.sendCommandEventStub = sandbox.stub(
+      telemetryService,
+      'sendCommandEvent'
+    );
+    hrtimeStub = sandbox.stub(process, 'hrtime');
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('Should stop function, show notification and send telemetry', async () => {
+    const srcUri = Uri.file(
+      path.join(getRootWorkspacePath(), 'functions', 'demoJavaScriptFunction')
+    );
+    const executor = new ForceFunctionStartExecutor();
+    const mockExecution = new MockExecution(executor.build(srcUri.fsPath));
+    const killExecutionStub = sandbox.stub(mockExecution, 'killExecution');
+    cliCommandExecutorStub.returns(mockExecution);
+    await forceFunctionStart(srcUri);
```

Comment:
Why don't we just stub FunctionService instead?

Reply:
Same reason as https://github.com/forcedotcom/salesforcedx-vscode/pull/2498#discussion_r488852960. We'd like to test the actual behavior of starting/stopping container over how we implement them.